example/c/api.h

```diff
@@ -0,0 +1,56 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+typedef struct DiplomatWriteable {
```

Comment:
suggestion: Ideally this stuff goes in a separate file, like diplomat.h

Reply:
should we automatically write out the file? or just have users copy it over from the repo?

Reply:
Unclear! I think probably automatically writing it out is the best, and we can provide tricks for namespacing/etc if necessary